(#2) Adds Initial Examples

JPRuskin opened the pull request:
## Description Of Changes
Adds an initial readme and some example deployments to the repository.

## Motivation and Context
These examples can be displayed and linked to on our docs page, referred to by teammembers, or contributed to by community members.

## Testing
- Tested importing several of the deployments to a recent install of CCM

### Operating Systems Testing
N/A

## Change Types Made

* ~[ ] Bug fix (non-breaking change).~
* [x] Feature / Enhancement (non-breaking change).
* ~[ ] Breaking change (fix or feature that could cause existing functionality to change).~
* [x] Documentation changes.
* ~[ ] PowerShell code changes.~

## Change Checklist

* [x] Requires a change to the documentation.
* [ ] Documentation has been updated.
    This needs to be merged before the documentation can link to it.
* ~[ ] Tests to cover my changes, have been added.~
* ~[ ] All new and existing tests passed?~
* ~[ ] PowerShell code changes: PowerShell v2 compatibility checked?~

## Related Issue

Fixes #1 
Resolves #2

st3phhays commented on images/SelectingTargetGroups.jpg:
Do you have Snaggit to add these red arrows?

JPRuskin replied:
I do, and can recreate them!

steviecoaster commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+The only deployment step contained upgrades (installing if not present) the `Firefox` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/firefox)).
+
+Assuming this package was available on your sources, you will only need to set the target groups for the deployment.
```
Tell them how to quickly get the package if they don't have it?

steviecoaster commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Usage
+
+The only deployment step contained upgrades (installing if not present) the `Firefox` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/firefox)).
```
I feel like this reads weird, but it could just be me?

steviecoaster commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+## Purpose
+
+This package is a basic example of deploying multiple packages.
```
```suggestion
This template is a basic example of deploying multiple packages.
```

st3phhays replied:
@steviecoaster Should it be `template` or `Deployment Plan` ? I've suggested this change in other files (before seeing this comment). I think in another file the word "deployment" is used.

steviecoaster replied:
@st3phhays I can agree on `Deployment Plan`

steviecoaster commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/installing-multiple-packages/Deployment.json#L1-L58
+
+This can scale neatly and easily, and you can generate larger deployments with more comprehensive suites of packages to deploy - even mixing basic and advanced deployments in order to provide additional customisation and configuration.
```
```suggestion
This can scale neatly and easily, and you can generate larger deployments with more comprehensive suites of packages to deploy - even mixing basic and advanced deployment steps in order to provide additional customisation and configuration.
```

steviecoaster replied:
I'm letting your `customisation` slide here ;) LOL

st3phhays commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Purpose
+
+This deployment is a basic example of deploying a package using a simple deployment step.
```
```suggestion
This Deployment Plan is a basic example of deploying a package using a simple deployment step.
```

st3phhays replied:
```suggestion
This Deployment Plan is a basic example of deploying a package using a simple deployment step.
```

st3phhays commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## Purpose
+
+This package is an example of deploying a package using an advanced deployment step, in this case to provide a package parameter during deployment.
```
```suggestion
This Deployment Plan is an example of deploying a package using an advanced Deployment Step, in this case to provide a package parameter during deployment.
```

st3phhays commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+## Purpose
+
+This package is a basic example of deploying multiple packages.
```
```suggestion
This Deployment Plan is a basic example of deploying multiple packages.
```

st3phhays commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+
+## Purpose
+
+This package is an example of upgrading the Chocolatey for Business client components.
```
```suggestion
These Deployment Plans are examples of upgrading the Chocolatey for Business client components.
```

st3phhays commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+## Purpose
+
+This package is a basic example of deploying a package, in this case `chocolatey.license`, on a schedule.
```
```suggestion
These Deployment Plans are basic examples of deploying a package, in this case `chocolatey.license`, on a schedule.
```

gep13 commented on README.md:
I am not sure why this file is showing as a showing binary, but it means that I can't leave comments on lines, but rather only on the file itself.

Couple things in this file though...

- CCM -> Chocolatey Central Management
- Chcoolatey -> Chocolatey

gep13 commented on deployments/installing-a-package-with-params/Deployment.json:
```diff
@@ -0,0 +1,22 @@
+      "Name": "Deploy Python to Agent Directory",
+      "IsPrivileged": true,
+      "PlanOrder": 1,
+      "Script": "choco install python312 --params=\"/InstallDir:C:\\BuildAgent\\tools\\python312\"\nexit $LASTEXITCODE",
```
Should we be applying [scripting best practices](https://docs.chocolatey.org/en-us/choco/commands/#scripting-integration-best-practices-style-guide) here?

JPRuskin replied:
Which one are you meaning?

gep13 replied:
Double quoting of option values `"''"`

gep13 commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json:
```diff
@@ -0,0 +1,34 @@
+      "Name": "Ensure Chocolatey 1.4.0",
+      "IsPrivileged": true,
+      "PlanOrder": 2,
+      "Script": "choco upgrade chocolatey --version 1.4.0 --source=\"'chocolatey.licensed'\"\nexit $LASTEXITCODE",
```
```suggestion
      "Script": "choco upgrade chocolatey --version=\"'1.4.0'\" --source=\"'chocolatey.licensed'\"\nexit $LASTEXITCODE",
```

gep13 commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json:
```diff
@@ -0,0 +1,34 @@
+      "Name": "Upgrade Chocolatey for Business Client Components",
+      "IsPrivileged": true,
+      "PlanOrder": 1,
+      "Script": "$delayInMinutes = 1\n\n# If using an internal repository to install Chocolatey Agent, replace `chocolatey.licensed` below\n# with the name or URL of your internally configured source.\nchoco upgrade chocolatey-agent --source 'chocolatey.licensed'\n\n# Ensure the deployment task registers as failed if the installation failed, and skip registering the\n# scheduled task.\nif ($LASTE ...
```
Should we be applying [scripting best practices](https://docs.chocolatey.org/en-us/choco/commands/#scripting-integration-best-practices-style-guide) here?

JPRuskin replied:
This is pretty close to directly taken from our docs. I'd hope that's okay, or ask what needs to be updated!

gep13 reviewed (dismissed):
I also think we need a complete review of these files to ensure that we are consistent in usage of:

- Deployment Plan
- Deployment Step

etc.

JPRuskin committed:
(#1) Updates Readme with Basic Description

Adds a basic readme with lightweight instructions on importing any old Deployment Plan, along with a friendly "please feel free to contribute" note.

pauby commented on deployments/installing-a-package-simple/Deployment.json:
Can you rename this file to `DeploymentPlan.json`?

JPRuskin replied:
I am doing this, but it might be worth noting for the next Chocolatey Central Management sprint that, seemingly, the default file name for a default-named deployment (plan) that is then exported is `Deployment $(Date) - Exported $(Date).json` and we could possibly change that to reduce (my) confusion. The UI also seems to refer to this concept as `Deployment` very consistently. As do [our docs](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments).

![image](https://github.com/chocolatey/ccm-deployment-examples/assets/1975761/c0dbd3e4-0dee-41c7-a76a-6c40b1fb8c27)

![image](https://github.com/chocolatey/ccm-deployment-examples/assets/1975761/2e170213-c5f4-4df1-9db4-ad37a56a3f30)

I should possibly push back on my initial acceptance of "Deployment Plan" in one of the other comments, as (looking at all this) I'm mildly regretting it.

Thoughts, @st3phhays @steviecoaster @gep13 ?

JPRuskin replied:
I gather this was discussed yesterday, and we'll be correcting everything else.

gep13 replied:
We are going to have a task in the next piece of upcoming work to standardize everything to use Deployment Plan. We will pick up this part as well.

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Purpose
+
+This Deployment Plan is a basic example of deploying a package using a simple deployment step.
```
```suggestion
This Deployment Plan is a basic example of deploying a package using a simple Deployment Step.
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+This Deployment Plan is a basic example of deploying a package using a simple deployment step.
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/installing-a-package-simple/Deployment.json#L1-L22
```
Does this embed it into this file?

JPRuskin replied:
Yes, though when it's in a permanant branch I intend to embed only the latest version of main, which will mean that we won't need to update it. Can't do that until _something_ is merged, though.

steviecoaster replied:
@JPRuskin this isn't rendering correctly, let's take it out.

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+## Usage
+
+The only Deployment Step in this Deployment Plan upgrades (installing if not present) the `Firefox` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/firefox)).
```
```suggestion
The only Deployment Step in this Deployment Plan upgrades, or installs if not present, the `Firefox` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/firefox)).
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+The only Deployment Step in this Deployment Plan upgrades (installing if not present) the `Firefox` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/firefox)).
+
+Assuming this package was available on your sources, you will only need to set the target groups for the deployment. If you don't have it available, you can download it using `choco download firefox --source="'https://community.chocolatey.org/api/v2/'"` and then upload it to your internal repository.
```
```suggestion
Assuming this package was available on your sources, you will only need to set the Target Groups for the Deployment Step. If you don't have it available, you can download it using `choco download firefox --source="'https://community.chocolatey.org/api/v2/'"` and then upload it to your internal repository.
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+You will need to set a target group for the deployment. 
+
+To do so:
```
```suggestion
You will need to set a Target Group for the Deployment Step:
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+To do so:
+
+- Mouse-over the deployment step and click the `Edit` button:
```
```suggestion
- Mouse-over the Deployment Step and click the `Edit` button.
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+
+- Click `Save`
+
+You are now ready to [move your deployment to Ready and run it](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#move-to-ready).
```
```suggestion
You are now ready to [move your Deployment Plan to Ready and run it](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#move-to-ready).
```

pauby commented on deployments/installing-a-package-with-params/Deployment.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "Name": "Deployment with Parameters",
```
```suggestion
  "Name": "Deployment Plan with Parameters",
```

pauby commented on deployments/installing-a-package-with-params/Deployment.json:
Can you rename this file to `DeploymentPlan.json`?

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+# Installing a Package with Parameters
```
```suggestion
# Installing a Package With Package Parameters
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## Purpose
+
+This Deployment Plan is an example of deploying a package using an advanced Deployment Step, in this case to provide a package parameter during deployment.
```
```suggestion
This Deployment Plan is an example of deploying a package using an Advanced Deployment Step, in this case to provide a package parameter during deployment.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/installing-a-package-with-params/Deployment.json#L1-L22
+
+We use the advanced step to do this because the basic deployment currently has no supported method to pass package parameters.
```
```suggestion
We use the Advanced Deployment Step to do this because the Basic Deployment Step does not allow package parameters to be used.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## Usage
+
+The only deployment step contained installs the `python312` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/python312)) if it's not already present.
```
```suggestion
The Deployment Step installs the `python312` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/python312)) if it's not already present.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+The only deployment step contained installs the `python312` package ([available on the Chocolatey Community Repository](https://community.chocolatey.org/packages/python312)) if it's not already present.
+
+Assuming this package was available on your sources (and you were happy with the install location), you will only need to set the target groups for the deployment.
```
```suggestion
Ensure the package is available on a configure source and set the Target Groups for the Deployment Step.
```

JPRuskin replied:
```suggestion
Ensure the package is available on a configured source and set the Target Groups for the Deployment Step.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+You will need to set a target group for the deployment. 
+
+To do so:
```
```suggestion
You will need to set a Target Group for the Deployment Step:
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+To do so:
+
+- Mouse-over the deployment step and click the `Edit` button:
```
```suggestion
- Mouse-over the Deployment Step and click the `Edit` button.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+
+- Click `Save`
+
+You are now ready to [move your deployment to Ready and run it](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#move-to-ready).
```
```suggestion
You are now ready to [move your Deployment Plan to Ready and run it](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#move-to-ready).
```

pauby commented on deployments/installing-multiple-packages/Deployment.json:
Can you rename this file to `DeploymentPlan.json`?

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+This Deployment Plan is a basic example of deploying multiple packages.
+
+Though Chocolatey CLI supports passing multiple package IDs in to a single `install` or `upgrade` command, CCM should use a separate deployment step for each so that per-computer reporting is easier to parse and action.
```
```suggestion
We recommend that you use a separate Deployment Step for each package to deploy to allow per-computer reporting.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/installing-multiple-packages/Deployment.json#L1-L58
+
+This can scale neatly and easily, and you can generate larger deployments with more comprehensive suites of packages to deploy - even mixing basic and advanced Deployment Steps in order to provide additional customization and configuration.
```
```suggestion
You can use this to generate larger Deployment Plans with more comprehensive suites of packages, even mixing Basic and Advanced Deployment Steps in order to provide additional configuration.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+## Usage
+
+This deployment is an example of deploying a set selection of software, for example, as a baseline for a company machine or department's requirements. The deployment could be scheduled to re-run against a specific group to ensure that the software is present and up to date.
```
```suggestion
This Deployment Plan is an example of deploying a selection of software such as a baseline for a company computer or to meet a department's requirements. The Deployment Plan could be scheduled to re-run against a specific group to ensure that the software is always present and up to date.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+This deployment is an example of deploying a set selection of software, for example, as a baseline for a company machine or department's requirements. The deployment could be scheduled to re-run against a specific group to ensure that the software is present and up to date.
+
+After importing the deployment, you will need to set the target group(s).
```
```suggestion
After importing the Deployment Plan, you will need to set the Target Groups.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+### Importing the Deployment
+
+- Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example)
```
If you change the name of the file, it will need updated in both places here.

pauby replied:
```suggestion
- Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example).
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+- Click `Browse` and select the downloaded json file  
+    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)
+
+- Click `Import`
```
```suggestion
- Login to your CCM instance

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` file.
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.
```

pauby commented on deployments/installing-a-package-with-params/README.md:
```diff
@@ -0,0 +1,35 @@
+- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
+    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
+
+- Click `Save`
```
```suggestion
- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```

pauby commented on deployments/installing-a-package-simple/README.md:
```diff
@@ -0,0 +1,31 @@
+- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
+    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
+
+- Click `Save`
```
```suggestion
- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+- Click `Import`
+
+For further details, please see the [documentation at docs.chocolatey.org](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```
```suggestion
See the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments) for more information.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+
+You will need to set a target group for the deployment. 
+
+To do so:
```
```suggestion
You will need to set a Target Group for the Deployment Step:
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,51 @@
+- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
+    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
+
+- Click `Save`
```
```suggestion
- Mouse-over the deployment step and click the `Edit` button.
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json:
```diff
@@ -0,0 +1,34 @@
+{
+  "Name": "Upgrade Chocolatey for Business Client to Latest 1.*",
```
```suggestion
  "Name": "Upgrade Chocolatey for Business Client to Latest 1.x",
```

pauby replied:
We use '1.x' everywhere rather than '1.*'.

pauby commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json:
```diff
@@ -0,0 +1,34 @@
+  "RequiresApproval": false,
+  "DeploymentSteps": [
+    {
+      "Name": "Upgrade to Latest 1.* Client Components",
```
```suggestion
      "Name": "Upgrade to Latest 1.x Client Components",
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+
+The second upgrades Chocolatey to the latest version.
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json#L1-L34
```
```suggestion
There are two example Deployment Plans here. They follow the instructions laid out on the [upgrading to Chocolatey v2](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#upgrade-using-chocolatey-central-management-deployments) page.

The first upgrades the Chocolatey for Business client components to the latest 1.x version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json#L1-L34

The second upgrades Chocolatey CLI to the latest version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json#L1-L34
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json:
```diff
@@ -0,0 +1,34 @@
+      "DeploymentStepGroupIds": []
+    },
+    {
+      "Name": "Ensure Chocolatey is Upgraded",
```
```suggestion
      "Name": "Ensure Chocolatey CLI is Upgraded",
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json:
```diff
@@ -0,0 +1,34 @@
+      "DeploymentStepGroupIds": []
+    },
+    {
+      "Name": "Ensure Chocolatey 1.4.0",
```
```suggestion
      "Name": "Ensure Chocolatey CLI 1.4.0",
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+
+## Usage
+
+You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.*` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```
```suggestion
You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+
+If you are upgraded to Chocolatey CLI version 2.0 or above, you can just import and run `Latest2x`.
+
+> Although the ScheduledTasks module is available on Windows Server 2012 R2, the chocolatey-agent service encounters an error when trying to import it. It is recommended to explore other options for the scheduled task if you're using Windows Server 2012 R2.
```
```suggestion
> Although the ScheduledTasks module is available on Windows Server 2012 R2, the `chocolatey-agent` service encounters an error when trying to import it. It is recommended to explore other options for the scheduled task if you're using Windows Server 2012 R2.
```

pauby replied:
What other options?

JPRuskin replied:
Jokingly: Upgrading to a more recent OS.

In more seriousness: We don't currently have company recommendations for any alternatives to this. I thought it prudent to include the warning, as I found it interesting when reading the docs, but would rather remove the warning than add recommendations for an SKU that had (initial) extended support end today (🎉). If we really want, I can investigate and create a `DeploymentPlan-2012R2Only.json` with an alternative? I think I have a few options available...

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+
+> Although the ScheduledTasks module is available on Windows Server 2012 R2, the chocolatey-agent service encounters an error when trying to import it. It is recommended to explore other options for the scheduled task if you're using Windows Server 2012 R2.
+
+After importing the deployments, you will need to set both the target groups. We strongly recommend testing this on a smaller group to ensure it works nicely in your environment.
+
+### Importing the Deployment
+
+- Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory
+
+- Login to your CCM instance
+
+- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
+    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)
+
+- Click `Browse` and select the downloaded json files  
+    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)
+
+- Click `Import`
+
+For further details, please see the [documentation at docs.chocolatey.org](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```
```suggestion
After importing the Deployment Plans, you will need to set the Target Groups for each. We strongly recommend testing this on a smaller group to ensure it works as expected in your environment.

### Importing the Deployment Plans

- Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory.

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  .
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` files.
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

JPRuskin replied:
```suggestion
After importing the Deployment Plans, you will need to set the Target Groups for each. We strongly recommend testing this on a smaller group to ensure it works as expected in your environment.

### Importing the Deployment Plans

- Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory.

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`.  
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` files.  
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,59 @@
+- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
+    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
+
+- Click `Save`
```
```suggestion
You will need to set a Target Group for the Deployment Steps:

- Mouse-over the Deployment Step and click the `Edit` button.
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```

pauby commented on deployments/upgrading-license-daily/Deployment-Weekly.json:
Can you rename this file to indicate it's a Deployment Plan?

pauby commented on deployments/upgrading-license-daily/Deployment.json:
Can you rename this file to indicate it's a Deployment Plan?

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+## Purpose
+
+These Deployment Plans are basic examples of deploying a package, in this case `chocolatey.license`, on a schedule.
```
```suggestion
These Deployment Plans are basic examples of deploying a package on a schedule. In this case `chocolatey.license`.
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+## Usage
+
+The only deployment step contained upgrades (installing if not present) the `chocolatey.license` package (which is we recommend creating when using C4B environments, to ensure you can push license updates out). If you don't currently have a license package, you can create one by following [the instructions available at docs.chocolatey.org](https://docs.chocolatey.org/en-us/c4b-environments/azure/ ...
+
+There is a second example deployment that would run weekly.
```
```suggestion
We recommend creating a `chocolatey.license` package when using Chocolatey for Business environments. This ensures you can push out license changes to your client computers.

The Deployment Step upgrades, or installs if not present, the `chocolatey.license` package . If you don't currently have a license package, you can create one by following [the instructions](https://docs.chocolatey.org/en-us/c4b-environments/azure/license-update#creating-a-new-license-package).

There is a second example Deployment Plan that would run weekly.
```

JPRuskin replied:
```suggestion
We recommend creating a `chocolatey.license` package when using Chocolatey for Business environments. This ensures you can push out license changes to your client computers.

The Deployment Step upgrades, or installs if not present, the `chocolatey.license` package. If you don't currently have a license package, you can create one by following [the instructions](https://docs.chocolatey.org/en-us/c4b-environments/azure/license-update#creating-a-new-license-package).

There is a second example Deployment Plan that would run weekly.
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-license-daily/Deployment-Weekly.json#L3-L5
+
+There are [more options available](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#recurring-deployments) for scheduling. You can see the required values for your current version of Chocolatey Central Management in the provided swagger document.
```
Where is the swagger document?

JPRuskin replied:
It is hard to provide an answer to this, as we only make it available on individual instances so the FQDN will change per-user. That said, I could link [here](https://docs.chocolatey.org/en-us/central-management/usage/api/#description) or add instructions here.

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+There are [more options available](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#recurring-deployments) for scheduling. You can see the required values for your current version of Chocolatey Central Management in the provided swagger document.
+
+After importing the deployment, you will need to set both the target groups (we recommend using the `All Computers` group) and the `Start Date Time`. You may also want to change the deployment name to something more meaningful.
```
```suggestion
After importing the Deployment Plans, you will need to set the Target Groups and `Start Date Time` for each. We recommend using the `All Computers` Target Group. You may also want to change the Deployment Plan name to something more meaningful.
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+After importing the deployment, you will need to set both the target groups (we recommend using the `All Computers` group) and the `Start Date Time`. You may also want to change the deployment name to something more meaningful.
+
+### Importing the Deployment
+
+- Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example)
+
+- Login to your CCM instance
+
+- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
+    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)
+
+- Click `Browse` and select the downloaded json file  
+    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)
+
+- Click `Import`
+
+For further details, please see the [documentation at docs.chocolatey.org](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```
```suggestion
### Importing the Deployment Plans

- Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example).

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`.
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` file.
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+### Setting the Start Date Time
+
+Before you can run the deployment, you must set the `Start Date Time` to a time at least 30 minutes in the future.
```
```suggestion
Before you can run the Deployment Plan, you must set the `Start Date Time` to a time at least 30 minutes in the future.
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+### Setting the Target Group
+
+You will also need to set a target group for the deployment. 
+
+To do so:
```
```suggestion
You will also need to set a Target Group for the Deployment Step:
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+To do so:
+
+- Mouse-over the deployment step and click the `Edit` button:
+    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)
+
+- Navigate to the `Select Target Groups` panel
+
+- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
+    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
+
+- Click `Save`
```
```suggestion
- Mouse-over the Deployment Step and click the `Edit` button.
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```

pauby reviewed (changes requested):
Deployments should be Deployment Plan or Deployment Steps. Both, along with Target Groups, are names of things and should be capitalized.

steviecoaster commented:
@JPRuskin It _looks_ like you've got everything fixed up here. Can you tidy up the commits on this PR and then we can merge it?

pauby commented:
@steviecoaster The README.md file needs to be updated with similar changes to the rest of the doc. I can't provide any comments on that, I'm _assuming_ because GitHub is seeing it as a binary file.

pauby commented:
As I mentioned, I can't provide any comments on the README.md file, so I'll add them here.

1. Heading 'Chocolatey Central Management Deployment Examples' should be 'Chocolatey Central Management Deployment Plan Examples'.
2. Bullet points need to have a full stop at the end.
3. 'target groups' should be 'Target Groups'.
4. For further details .... we don't need docs.chocolatey.org in there as it's already a clickable URL.
5. Contributing -> 'target groups' should be 'Target Groups'
6. Contributing -> bullet point 2: 'readme' should be 'README'.
7. Contributing -> bullet points should all end with full stops.

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,49 @@
+
+- Download the `DeploymentPlan.json` from this directory (or the `DeploymentPlan-Weekly.json` example).
+
+- Login to your CCM instance
```
```suggestion
- Login to your Chocolatey Central Management instance.
```

pauby commented on deployments/installing-multiple-packages/README.md:
```diff
@@ -0,0 +1,49 @@
+
+- Login to your CCM instance
+
+- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
```
```suggestion
- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment` .
```

pauby commented on deployments/upgrading-chocolatey-for-business-clients/README.md:
```diff
@@ -0,0 +1,57 @@
+
+## Usage
+
+You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```
```suggestion
You should import both the `Latest1x` and `Latest2x` Deployment Plans if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```

pauby commented on deployments/upgrading-license-daily/README.md:
```diff
@@ -0,0 +1,61 @@
+
+Before you can run the Deployment Plan, you must set the `Start Date Time` to a time at least 30 minutes in the future.
+
+To do so, click on the clock icon next to the `Start Date Time` field. When you have selected an appropriate time, hit `Save`.
```
```suggestion
To do so, click on the clock icon next to the `Start Date Time` field. When you have selected an appropriate time, click `Save`.
```

JPRuskin commented:
> As I mentioned, I can't provide any comments on the README.md file, so I'll add them here.

FWIW, this should be sorted in all future PRs. Just this one where we're going from the template's UTF16 to UTF8 (like all our other md using repos, as far as I can tell).

Hopefully all of the above have been addressed, now.

JPRuskin committed:
(#2) Adds Initial Examples

This commit adds the initial examples to the repository.

Also updates to "Deployment Plans" rather than "Deployment(s)"
